fix: IndexError if values exceeded scale_discrete palette size

TyberiusPrime opened the pull request:
This plot would fail with an IndexError in map (and not just a warning for omitted values due to palette limits):

```python
  df = pd.DataFrame({
        # not that it's the second c that triggered a bug in scale_discrete.map
        'x': pd.Categorical(['a', 'b', 'c', 'c']),
        'y': [0, 1, 2, 2]})
    p = (ggplot(df, aes('x', 'y', color='x', shape='x'))
         + geom_jitter()
         + scale_color_manual(['red', 'blue'])
         ).save('test.png')
```

This PR fixes it by replacing a complicated numpy 'insert NaNs at indices' solution with a call to pandas.Series.reindex() which helpfully replaces non-existent indices with the NaNs we need.

has2k1 commented on plotnine/tests/test_scale_internals.py:
```diff
@@ -645,3 +645,16 @@ def test_layer_with_only_infs():
+def test_discrete_scale_exceeding_maximum_number_of_values():
+    df = pd.DataFrame({
+        # not that it's the second c that triggered a bug in scale_discrete.map
+        'x': pd.Categorical(['a', 'b', 'c', 'c']),
```
Use `['c', 'a', 'c', 'b', 'c']`. If the logic, ever changes this is trickier.

has2k1 commented:
Also, keep the commit title [50 characters or less](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)

TyberiusPrime commented:
I fixed the commit title and switched to your test suggestion.

As for the actual code, I did some profiling:
![image](https://user-images.githubusercontent.com/1257580/63415944-ce31f200-c3fe-11e9-9439-eb0f276151f6.png)

pandas is the one I proposed initially, list_comp is your suggestion, 
clip is the one I've pushed now.

I'm currently calculating with sample size = 100k...

TyberiusPrime commented:
Ok, 100k results are in - clip still beats pandas at that size, so let's stick with that.


![image](https://user-images.githubusercontent.com/1257580/63420240-962ead00-c406-11e9-9dbe-f66194acdb02.png)

has2k1 commented:
Thanks. Revealing profiling results, kind of what I would have expected.
